clients/client-go/tcgithub/types.go

```diff
@@ -110,6 +112,136 @@ type (
+		//   * "unassigned"
+		//   * "unlabeled"
+		Action string `json:"action"`
+
+		// Additional data to be mixed to the mocked event object.
+		// This can be used to set some specific properties of the event or override the existing ones.
+		// For example:
+		//   "ref": "refs/heads/main"
+		//   "before": "000"
+		//   "after": "111"
+		//
+		// Additional properties allowed
+		Overrides json.RawMessage `json:"overrides,omitempty"`
+
+		// Possible values:
+		//   * "github-pull-request"
+		//   * "github-pull-request-untrusted"
+		Type string `json:"type"`
+	}
+
+	// Emulate one of the github events with mocked payload.
+	// Some of the events have sub-actions, that can be specified.
+	// Event type names follow the `tasks_for` naming convention.
+	PushEvents struct {
+
+		// Additional data to be mixed to the mocked event object.
+		// This can be used to set some specific properties of the event or override the existing ones.
+		// For example:
+		//   "ref": "refs/heads/main"
+		//   "before": "000"
+		//   "after": "111"
+		//
+		// Additional properties allowed
+		Overrides json.RawMessage `json:"overrides,omitempty"`
+
+		// Possible values:
+		//   * "github-push"
+		//   * "github-tag-push"
+		Type string `json:"type"`
+	}
+
+	// Emulate one of the github events with mocked payload.
+	// Some of the events have sub-actions, that can be specified.
+	// Event type names follow the `tasks_for` naming convention.
+	ReleaseEvents struct {
+
+		// Possible values:
+		//   * "published"
+		//   * "unpublished"
+		//   * "created"
+		//   * "edited"
+		//   * "deleted"
+		//   * "prereleased"
+		//   * "released"
+		Action string `json:"action"`
+
+		// Additional data to be mixed to the mocked event object.
+		// This can be used to set some specific properties of the event or override the existing ones.
+		// For example:
+		//   "ref": "refs/heads/main"
+		//   "before": "000"
+		//   "after": "111"
+		//
+		// Additional properties allowed
+		Overrides json.RawMessage `json:"overrides,omitempty"`
+
+		// Possible values:
+		//   * "github-release"
+		Type string `json:"type"`
+	}
+
+	// Render .taskcluster.yml for one of the supported events.
+	RenderTaskclusterYmlInput struct {
+
+		// The contents of the .taskcluster.yml
+		Body string `json:"body"`
+
+		// Emulate one of the github events with mocked payload.
+		// Some of the events have sub-actions, that can be specified.
+		// Event type names follow the `tasks_for` naming convention.
+		//
+		// One of:
+		//   * PushEvents
+		//   * PullRequestEvents
+		//   * ReleaseEvents
+		FakeEvent json.RawMessage `json:"fakeEvent"`
+
+		// Syntax:     ^[-a-zA-Z0-9]{1,39}$
+		Organization string `json:"organization,omitempty"`
+
+		// Syntax:     ^[-a-zA-Z0-9_.]{1,100}$
+		Repository string `json:"repository,omitempty"`
+	}
+
+	// Rendered .taskcluster.yml output.
+	RenderTaskclusterYmlOutput struct {
+
+		// Scopes that will be used by the github client to create tasks.
```

Comment:
they actually do! Depending on type of event, branch, and tag version, they would produce different set of scopes. 

The reason I decided to include those is because this is somewhat hidden to the user and not obvious that different there can be different scopes.